bears/vcs/VCSCommitBear.py

```diff
@@ -0,0 +1,126 @@
+import abc
+import re
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.results.HiddenResult import HiddenResult
+from coalib.misc.Shell import run_shell_command
+from coala_utils.decorators import (enforce_signature, generate_ordering,
+                                    generate_repr)
+
+
+@generate_repr(('id', hex),
+               'origin',
+               'raw_commit_message',
+               'commit_sha',
+               'commit_type',
+               'modified_files',
+               'added_files',
+               'deleted_files',
+               'message')
+@generate_ordering('raw_commit_message',
+                   'commit_sha',
+                   'commit_type',
+                   'modified_files',
+                   'added_files',
+                   'deleted_files',
+                   'severity',
+                   'confidence',
+                   'origin',
+                   'message_base',
+                   'message_arguments',
+                   'aspect',
+                   'additional_info',
+                   'debug_msg')
+class CommitResult(HiddenResult):
+
+    @enforce_signature
+    def __init__(self, origin,
+                 raw_commit_message: str,
+                 commit_sha: str,
+                 commit_type: list,
+                 modified_files: list,
+                 added_files: list,
+                 deleted_files: list):
+
+        HiddenResult.__init__(self, origin, '')
+        self.raw_commit_message = raw_commit_message
+        self.commit_sha = commit_sha
+        self.commit_type = commit_type
+        self.modified_files = modified_files
+        self.added_files = added_files
+        self.deleted_files = deleted_files
+        self.message = 'HEAD commit information'
+
+
+class VCSCommitBear(GlobalBear):
+    __metaclass__ = abc.ABCMeta
+    LANGUAGES = {'Git'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    @abc.abstractmethod
+    def get_head_commit(self):
+        """
+        Return the commit message from the head commit
+        """
+
+    def analyze_git_commit(self, head_commit):
+        raw_commit_message = head_commit
+
+        commit_sha = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+
+        commit_type = []
+
+        head_commit = head_commit.strip('\n')
+
+        if re.search(r'\[ci skip\]|\[skip ci\]', head_commit):
+            commit_type.append('ci_skip_commit')
+
+        get_parent_commits = 'git log --pretty=%P -n 1 ' + commit_sha
+        parent_commits = run_shell_command(get_parent_commits)[0]
+        parent_commits_list = parent_commits.split('\n')
```

Comment:
I think this is the reason why `merge_commit` always appears in the testcase. I think you should split it with a space, not a newline
```
> git log --pretty=%P -n 1 (git rev-parse HEAD)
00c07fb1d4eeaaaa4cd76d6650b5ccfbb0f40be8 38536dfc0a633bb40b99013716ac26642ac17c84
```